src/main/java/com/solvd/pages/ResultPage.java

```diff
@@ -17,11 +19,12 @@ public class ResultPage {
     private WebElement resultsNumberOnPage;
 
     public ResultPage(WebDriver driver){
-        this.driver = driver;
-        PageFactory.initElements(driver, this);
+        super(driver);
     }
 
     public boolean isResultListEmpty(){
+        WebDriverWait wait = new WebDriverWait(this.getDriver(), Duration.ofSeconds(20));
+        wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//span[@class ='a-size-medium a-color-base a-text-normal']")));
```

Comment:
in wait you may try to use:
1) ExpectedConditions.visibilityOfAllElements(resultList) - to wait for all elements
2) ExpectedConditions.visibilityOf(resultList.get(0)) - to wait just for first element
so you don't need to copy the xpath.